src/search/evaluators/sub_evaluator.cc

```diff
@@ -0,0 +1,98 @@
+#include "sub_evaluator.h"
+
+#include "../evaluation_context.h"
+#include "../evaluation_result.h"
+#include "../option_parser.h"
+#include "../plugin.h"
+
+#include <cstdlib>
+#include <sstream>
+#include <cmath>
+
+using namespace std;
+namespace sub_evaluator {
+    SubEvaluator::SubEvaluator(const Options &opts)
+            : evaluator(opts.get < shared_ptr < Evaluator >> ("eval")),
+              w(opts.get<double>("weight")),
+              t(opts.get<EvaluatorType>("type")) {
+    }
+
+    SubEvaluator::SubEvaluator(const shared_ptr <Evaluator> &eval, double weight, EvaluatorType type)
+            : evaluator(eval), w(weight), t(type) {
+    }
+
+    SubEvaluator::~SubEvaluator() {
+    }
+
+    bool SubEvaluator::dead_ends_are_reliable() const {
+        return evaluator->dead_ends_are_reliable();
+    }
+
+    EvaluationResult SubEvaluator::compute_result(
+            EvaluationContext &eval_context) {
+        // Note that this produces no preferred operators.
+        int g = eval_context.get_g_value();
+        EvaluationResult result;
+        int value = eval_context.get_evaluator_value_or_infinity(evaluator.get());
+        if (value != EvaluationResult::INFTY) {
+            if (t == WA) {
+                value += ((1 / w) * g);
```

Comment:
I think it would be clearer to use 

    int h = eval_context.get_evaluator_value_or_infinity(evaluator.get());
    int value = -1;

above and 

    value = g/w + h;

here. Similarly, you can then use `h` instead of `value` in the other formulas.